### DISCUSSION TITLE
Heroku Config

### DISCUSSION THREAD

**Discussion by gettract** [sidekiq/sidekiq#5946] [Q&A]
On this page in the wiki: https://github.com/sidekiq/sidekiq/wiki/Using-Redis it says that you can:

```
heroku config:set REDIS_PROVIDER=REDISCLOUD_URL
```

I can't get this to work. Example:

```
heroku config:set TEST=REDISCLOUD_URL

heroku config
TEST:                     REDISCLOUD_URL

heroku run rails c

> ENV['REDISCLOUD_URL']
"redis://default:XXXXXXXX@redis-160XXX01.c246.us-east-1-4.ec2.cloud.redislabs.com:16001"

> ENV['TEST']                                                                                                                 
=> nil
```

I know this isn't a sidekiq problem specifically, but you may want to edit the wiki if heroku has changed behavior.

**Comment by davidwessman** [comment]
It goes the other way around, if you have maybe two providers on Heroku that you want to switch between:

```
REDIS_URL=redis://...
REDISCLOUD_URL=redis://...
```

then you can tell Sidekiq which of these to use by setting:
`REDIS_PROVIDER=REDIS_URL`
or
`REDIS_PROVIDER=REDISCLOUD_URL`

e.g `REDIS_PROVIDER` should be set to the name of the variable you want to read from.

**Reply by jess** [reply]
Thanks David, but how do you set the `REDIS_PROVIDER` env var if you don't know it in advance. If you're using something like a review app on heroku, then the redis cloud value is going to be different on each app. So unless we tell sidekiq to reference the redis cloud var specifically or manually create `REDIS_PROVIDER=redis://` then it won't know.

**Reply by davidwessman** [reply]
If your review apps have the `REDISCLOUD_URL` value set, you can use this literally:
`REDIS_PROVIDER=REDISCLOUD_URL`
and it should work.

**Reply by davidwessman** [reply]
You do not need to know the actual redis://-url, only the name of the variable

**Reply by jess** [reply]
Sorry to be so difficult, but where do you put `REDIS_PROVIDER=REDISCLOUD_URL`? Is that set in the rails app, if so where? Heroku ENV?

**Reply by davidwessman** [reply]
We have it in our app.json, like

```
{
  "env": {
    "REDIS_PROVIDER": {
      "value": "BONSAI_URL"
    }
  }
}
```

I omitted all other values, see more examples at Heroku.

https://devcenter.heroku.com/articles/app-json-schema#example-app-json

Then I think you can use it for review apps.

**Reply by davidwessman** [reply]
There is also some setting for adding env-variables to review apps via the pipeline settings.
